tests/unit/components/ObsDetails/ObsDetails.test.js

```diff
@@ -99,41 +110,62 @@ describe( "Observation with no evidence", () => {
   } );
 } );
 
-test( "renders data tab on button press", async ( ) => {
-  const { getByText, findByTestId } = renderComponent( <ObsDetails /> );
-  const button = await findByTestId( "ObsDetails.DataTab" );
+describe( "activity tab", ( ) => {
+  test( "navigates to observer profile on button press", async ( ) => {
+    const { findByTestId } = renderComponent( <ObsDetails /> );
 
-  fireEvent.press( button );
-  expect( getByText( mockObservation.description ) ).toBeTruthy( );
-} );
+    fireEvent.press( await findByTestId( "ObsDetails.currentUser" ) );
+    expect( mockNavigate )
+      .toHaveBeenCalledWith( "UserProfile", { userId: mockObservation.user.id } );
+  } );
 
-test( "navigates to observer profile on button press", async ( ) => {
-  const { findByTestId } = renderComponent( <ObsDetails /> );
+  test( "navigates to identifier profile on button press", async ( ) => {
+    const { findByTestId } = renderComponent( <ObsDetails /> );
 
-  fireEvent.press( await findByTestId( "ObsDetails.currentUser" ) );
-  expect( mockNavigate )
-    .toHaveBeenCalledWith( "UserProfile", { userId: mockObservation.user.id } );
-} );
+    fireEvent.press(
+      await findByTestId(
+        `ObsDetails.identifier.${mockObservation.identifications[0].user.id}`
+      )
+    );
+    expect( mockNavigate ).toHaveBeenCalledWith( "UserProfile", {
+      userId: mockObservation.identifications[0].user.id
+    } );
+  } );
 
-test( "navigates to identifier profile on button press", async ( ) => {
-  const { findByTestId } = renderComponent( <ObsDetails /> );
+  test( "navigates to taxon details on button press", async ( ) => {
+    const { findByTestId } = renderComponent( <ObsDetails /> );
 
-  fireEvent.press(
-    await findByTestId(
-      `ObsDetails.identifier.${mockObservation.identifications[0].user.id}`
-    )
-  );
-  expect( mockNavigate ).toHaveBeenCalledWith( "UserProfile", {
-    userId: mockObservation.identifications[0].user.id
+    fireEvent.press( await findByTestId( `ObsDetails.taxon.${mockObservation.taxon.id}` ) );
+    expect( mockNavigate ).toHaveBeenCalledWith( "TaxonDetails", {
+      id: mockObservation.taxon.id
+    } );
+  } );
+  test( "shows network error image if user is offline", ( ) => {
+    useIsOnline.mockImplementation( ( ) => false );
+    renderComponent( <ObsDetails /> );
+    const noInternet = screen.queryByRole( "image", { name: "network-check" } );
+    expect( noInternet ).toBeTruthy( );
+    expect( screen.queryByTestId( "PhotoScroll.photo" ) ).toBeNull( );
   } );
 } );
 
-test( "navigates to taxon details on button press", async ( ) => {
-  const { findByTestId } = renderComponent( <ObsDetails /> );
+describe( "data tab", ( ) => {
+  test( "renders data tab content when DATA tab is pressed", ( ) => {
+    const { getByText } = renderComponent( <ObsDetails /> );
+    const dataTab = screen.getByText( /DATA/ );
+    fireEvent.press( dataTab );
```

Comment:
Let's also check that the data tab is falsey before navigating to it.

Reply:
You lost me here. Why would the data tab be falsy while it's visible in the DOM and available for pressing?